pandas/core/indexes/base.py

```diff
@@ -2955,10 +2960,11 @@ def difference(self, other):
         label_diff = np.setdiff1d(np.arange(this.size), indexer,
                                   assume_unique=True)
         the_diff = this.values.take(label_diff)
-        try:
-            the_diff = sorting.safe_sort(the_diff)
-        except TypeError:
-            pass
```

Comment:
can you add some tests in the index tests to exercise this (prob just parameterize the parameter in the tests)